trpcultivate_phenoshare/src/Plugin/TripalImporter/TripalCultivatePhenoshareImporter.php

```diff
@@ -357,24 +414,45 @@ public function formValidate($form, &$form_state) {
        
         // All values will be accessible to every instance of the validator Plugin.
         $project = $form_state_values['project'];
-        // @TODO: this will be a select field.
-        $genus = '#';
+        $genus = $form_state_values['genus'];
         $file = $form_state_values['file_upload'];
+        // @TODO: load headers in validators that require reference to the headers.
+        $headers = [];
 
         if ($stage == 1) {
-          $scope = 'PROJECT';
-          // Create instance of the scope-specific plugin and perform validation.
-          $validator = $manager->getValidatorIdWithScope($scope);
-          $instance = $manager->createInstance($validator);
-          $instance->loadAssets($project, $genus, $file);
-          
-          // Perform Project Level validation.
-          $validation[ $scope ] = $instance->validate();
+          $scopes = ['PROJECT', 'GENUS'];
           
-          // Save validation result.
-          $storage = $form_state->getStorage();
-          $storage[ $this->validation_result ] = $validation;
-          $form_state->setStorage($storage);
+          foreach($scopes as $scope) {
+            // Create instance of the scope-specific plugin and perform validation.
+            $validator = $manager->getValidatorIdWithScope($scope);
+            $instance = $manager->createInstance($validator);
+            
+            // Set other validation level to upcoming/todo if a validation failed.
+            $skip = ($failed_validator > 0) ? 1 : 0;
+            
+            // Load values.
+            $instance->loadAssets($project, $genus, $file, $headers, $skip);
+            
+            // Perform Project Level validation.
+            $validation[ $scope ] = $instance->validate();
+
+            // Save validation result.
+            $storage = $form_state->getStorage();
+            $storage[ $this->validation_result ] = $validation;
```

Comment:
I feel like this will only store the validation for the last scope 🤔 Since the key would be the same in $storage and this is a loop for multiple scopes... It's looks like $validation is keyed by scope so maybe this just needs to be an array merge if it's already set?

Reply:
interesting too 🤔 , but then it's unique in that none of the scope is duplicated as reported by validation result

![Screen Shot 2023-11-09 at 2 44 23 PM](https://github.com/TripalCultivate/TripalCultivate-Phenotypes/assets/15472253/86e56bf8-1d33-41b6-808b-1096b8fe83dd)

Maybe a good idea is keep building the array and when completed write/save it once - might improve the speed too.

Reply:
I have revised this section of the code